custom_components/mammotion/entity.py

```diff
@@ -24,6 +24,7 @@ def __init__(self, coordinator: MammotionDataUpdateCoordinator, key: str) -> Non
             name=coordinator.device_name,
             # ToDo: To add in once betterproto is fixed
             # sw_version=coordinator.device.luba_msg.net.toapp_devinfo_resp.resp_ids.get(0, {}).get('info', "Loading..."),
+            sw_version=coordinator.device.raw_data["net"]["toapp_devinfo_resp"]["resp_ids"][0]["info"], # raw_data is a temp workaround until betterproto is fixed
```

Comment:
this will crash if there is nothing in the array. Your going to need to do something safer.

Reply:
Yolo